packages/nl-design-system-unstable/__docs__/form-patterns.md

```diff
@@ -1,139 +1,270 @@
+
+De verschillende vormen van invoer elementen en hun specifieke richtlijnen vindt je in de componenten-bibliotheek.
+
+Voor ingevulde/al bekende informatie
+
+Als informatie vooraf al bekend is, bijvoorbeeld een adres of telefoonnummer, vul deze dan in maar laat de bezoeker deze informatie wel wijzigen.
+
+Dit kan op twee manieren:
+
+1. Het vooraf invullen van een formulierveld. Hierdoor heeft een gebruiker de mogelijkheid deze informatie direct aan te passen. Dit is uiteraard alleen van toepassing als de bezoeker is ingelogd.
+2. Het tonen van de informatie (_‘read-only’_) met een functie om deze te wijzigen na het klikken op een button.
+
+## Hulp
+
+### Start
+
+Als een gebruiker informatie nodig heeft die zij niet zomaar voor handen zou kunnen hebben, zoals bijvoorbeeld een identiteitsbewijs, voeg dan een overzicht toe wat zij nodig heeft om een formulier volledig in te vullen. Zo komt een gebruiker niet voor verrassingen te staan.
+
+### Placeholder tekst
+
+Optionele placeholder tekst kan helpen bij het bepalen van het juiste gegevensformaat dat voor een bepaalde invoer wordt gevraagd.
+
+Placeholder tekst dient leesbaar te zijn en daarom een juiste contrastverhouding hebben. Omdat gebruikers de placeholder tekst kunnen verwarren met ingevoerde tekst, is het echter raadzaam om placeholder tekst te vermijden.
+
+### Help en feedback tekst
+
+Optionele help of feedback tekst kan dynamisch zijn, bijvoorbeeld wanneer een veld onjuist is ingevuld en je na validatie wilt uitleggen hoe deze fout gecorrigeerd kan worden.
+
+Overweeg om extra help tekst toe te voegen wanneer:
+
+* Je een bezoeker vraagt om minder bekende informatie.
+* Een bezoeker zich af zou kunnen vragen waarom om specifieke informatie gevraagd wordt.
+* Een bezoeker mogelijk bezorgd kan zijn over de veiligheid of privacy van bepaalde gegevens.
+
+### Optionele en verplichte invoervelden
+
+Overweeg om optionele vragen aan het einde van het formulier te stellen, tenzij deze echt bij een bepaald onderwerp/groep horen.
+
+Als niet alle velden verplicht zijn, maak dan duidelijk welke velden dat wel zijn.
+
+Als de meeste invoer op een formulier verplicht is, geef dan enkel diegene aan die optioneel zijn. Als de meeste invoer op een formulier optioneel is, geef dan enkel diegene aan die verplicht zijn.
+
+De beste manier om dit aan te geven is met een tekst zodat deze ook door schermlezers wordt aangekondigd. Zijn de meeste velden vereist in een formulier? Laat dan alleen een indicatie zien voor de optionele velden. Zorg er echter voor dat dit patroon consistent is in de hele interface.
+
+De duidelijkste manier is door aan het tekstlabel van het betreffende element respectievelijk de tekst ‘(optioneel)’ of ‘(verplicht)’ toe te voegen. Dit zorgt er voor dat voorleessoftware deze makkelijk kan interpreteren. Eventueel kan een asterisk (*) geplaatst worden bij verplichte velden. Zorg in dat geval echter wel dat ook voorleessoftware deze kan interpreteren.
```

Comment:
Dit is een lastige: je moet ook weer vermijden dat "verplicht" dubbel wordt aangekondigd. Bijvoorbeeld:
```html
<label for="user">Gebruikersnaam (verplicht)</label>
<input id="user" required type="text">
```
In bovenstaande voorbeeld wordt zowel "verplicht" voorgelezen uit de tekst als door het `required` attribuut.

Reply:
Ja goed punt! Dat pleit er voor om alleen optionele stappen te voorzien van deze indicator. Ik ga kijken hoe dit te verbeteren is.